scripts/generate_config_tests.py

```diff
@@ -88,12 +87,8 @@ def dependencies_of_setting(cfg: config_common.Config,
         if name.startswith('MBEDTLS_CIPHER_PADDING_'):
             return 'MBEDTLS_CIPHER_C:MBEDTLS_CIPHER_MODE_CBC'
         if name.startswith('MBEDTLS_PK_PARSE_EC_'):
-            #temporary solution to determine correct dependency macros between 3.6 and 4.0
-            #see issue #51 in mbedtls-framework
-            if build_tree.is_mbedtls_3_6():
-                return 'MBEDTLS_PK_C:MBEDTLS_PK_HAVE_ECC_KEYS'
-            else:
-                return 'MBEDTLS_PK_C:PSA_WANT_KEY_TYPE_ECC_PUBLIC_KEY'
+            return 'MBEDTLS_PK_C:' + test_case.psa_or_3_6_feature_macro(
+                'PSA_WANT_KEY_TYPE_ECC_PUBLIC_KEY', '')
```

Comment:
As a matter of principle, the domain should be `USE_PSA` for PK, even if the implementation will disregard that.